File: 프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.py
def solution(a, b):
    
    return sum([i for i in range(a, b + 1)] if a < b else [i for i in range(b, a + 1)])

# 다른 풀이
if a > b:
    a, b = b, a
return sum(range(a, b + 1))

# 다른 풀이
return (abs(a-b)+1)*(a+b)//2

# 다른 풀이
return sum(range(min(a, b), max(a, b)+1))

File: 프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.java
class Solution {
    public long solution(int a, int b) {
        long answer = 0;
        
        if(a > b) {
            int temp = a;
            a = b;
            b = temp;
        }
        
        for(int i = a; i <= b; i++) {
            answer += i;
        }
        
        return answer;
    }
}

// 다른 풀이
 public long solution(int a, int b) {
    return sumAtoB(Math.min(a, b), Math.max(b, a));
}

private long sumAtoB(long a, long b) {
    return (b - a + 1) * (a + b) / 2;
}

// 다른 풀이
long answer = 0;
for (int i = ((a < b) ? a : b); i <= ((a < b) ? b : a); i++) 
  answer += i;

return answer;
